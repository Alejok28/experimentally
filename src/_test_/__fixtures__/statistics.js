export const statistics = {
  "viewCount": "28640082",
  "likeCount": "556518",
  "dislikeCount": "6249",
  "favoriteCount": "0",
  "commentCount": "12663"
}