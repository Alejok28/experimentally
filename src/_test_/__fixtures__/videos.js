export const videos = [
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/n8La5FHPnJZ9SjefY7NHLK55KlY\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "z2VZ_OGz0RU"
    },
    "snippet": {
      "publishedAt": "2017-03-22T10:01:08.000Z",
      "channelId": "UCftwRNsjfRo08xYE31tkiyw",
      "title": "Ryan Reynolds &amp; Jake Gyllenhaal Answer the Web&#39;s Most Searched Questions | WIRED",
      "description": "'Life' stars Ryan Reynolds and Jake Gyllenhaal waste some time with WIRED and answer the Internet's most searched questions about themselves. Does Ryan ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/z2VZ_OGz0RU/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/z2VZ_OGz0RU/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/z2VZ_OGz0RU/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "WIRED",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/KPeWEEZzKXm-pfv-1ywpZ2WOtPE\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "F9CrRG6j2SM"
    },
    "snippet": {
      "publishedAt": "2019-08-05T09:00:02.000Z",
      "channelId": "UCfkXDY7vwkcJ8ddFGz8KusA",
      "title": "[M/V] SEVENTEEN(세븐틴) - HIT",
      "description": "13명의 뚜렷한 음악 색을 바탕으로 세븐틴 표 노래와 무대를 전 세계에 알리며 독보적인 독창성을 보여주고 있는 자체 제작 아이돌 세븐틴이 또...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/F9CrRG6j2SM/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/F9CrRG6j2SM/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/F9CrRG6j2SM/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "SEVENTEEN",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/id3uz64VI_Frb6lxlzAYDG8yCuw\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "A2FsgKoGD04"
    },
    "snippet": {
      "publishedAt": "2017-05-25T21:58:58.000Z",
      "channelId": "UCRijo3ddMTht_IHyNSNXpNQ",
      "title": "Real Life Trick Shots | Dude Perfect",
      "description": "Trick shots should be an everyday thing! Thanks to Kingsford for sponsoring this video! ▻ Click HERE to subscribe to Dude Perfect! http://bit.ly/SubDudePerfect ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/A2FsgKoGD04/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/A2FsgKoGD04/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/A2FsgKoGD04/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Dude Perfect",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/mGGEbULKJWR5g9X1nJ2MGKv0MKo\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "t_qn-f7XfJo"
    },
    "snippet": {
      "publishedAt": "2019-04-26T20:58:47.000Z",
      "channelId": "UCte53PewI8_jYpBYn071MqQ",
      "title": "Sech - Otro Trago ft. Darell (Video Oficial)",
      "description": "Escúchalo en Spotify: https://open.spotify.com/track/1Ej96GIBCTvgH7tNX1r3qr?si=xhlKQ1ZGR6iT5xnErFxojA Producido por Cinema Entertainment Sigue a ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/t_qn-f7XfJo/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/t_qn-f7XfJo/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/t_qn-f7XfJo/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Sech",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/1szGxIC2JpHOZayLDDyWY1wRMRg\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "VRJmcxCrAOA"
    },
    "snippet": {
      "publishedAt": "2018-01-22T22:55:24.000Z",
      "channelId": "UCRijo3ddMTht_IHyNSNXpNQ",
      "title": "Real Life Trick Shots 2 | Dude Perfect",
      "description": "Trick shots should be an everyday thing! Thanks to Kay Jewelers for sponsoring this video! Win Valentine's Day by going to http://dudeperfect.com/KayJewelers ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/VRJmcxCrAOA/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/VRJmcxCrAOA/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/VRJmcxCrAOA/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Dude Perfect",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/0WX426ECapOMYBfaUd1p8CnScQ8\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "rtgY1q0J_TQ"
    },
    "snippet": {
      "publishedAt": "2014-09-22T23:08:57.000Z",
      "channelId": "UCUsN5ZwHx2kILm84-jPDeXw",
      "title": "Key &amp; Peele - Gay Wedding Advice",
      "description": "In preparation for their cousin's marriage to another man, a family asks a gay man questions about what the wedding ceremony will be like and for \"advice\" about ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/rtgY1q0J_TQ/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/rtgY1q0J_TQ/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/rtgY1q0J_TQ/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Comedy Central",
      "liveBroadcastContent": "none"
    }
  }
]