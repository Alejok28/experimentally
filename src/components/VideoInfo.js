import React from 'react';
import { makeStyles } from '@material-ui/styles';
import PropTypes from 'prop-types';

// Components
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Typography from '@material-ui/core/Typography';

// Icons
import Like from '@material-ui/icons/ThumbUp';
import Dislike from '@material-ui/icons/ThumbDown';
import Comments from '@material-ui/icons/Comment';

// Others
import { escapeHtml } from '../utils/functions';

const useStyles = makeStyles({
  root: {
    margin: '50px 0'
  },
  container:{ 
    height: '50vh'
  },
  actions: {
    display: 'flex', justifyContent: 'space-between'
  },
  error: {
    textAlign: 'center',
    margin: '30px'
  }
});

export default function VideoInfo({ snippet, statistics }) {
  const classes = useStyles();
  return (
    <Card>
      <CardContent>
        <Typography gutterBottom variant="h5" component="h2">
          {escapeHtml(snippet.title)}
        </Typography>

        <Typography variant="body2" color="textSecondary" component="p">
          {snippet.description}
        </Typography>
      </CardContent>
      {statistics && (
        <CardActions className={classes.actions}>
          <Like color="secondary" />{statistics.likeCount}
          <Dislike color="error" />{statistics.dislikeCount}
          <Comments color="disabled" />{statistics.commentCount}
        </CardActions>
      )}
    </Card>
  );
}



VideoInfo.propTypes = {
  snippet: PropTypes.object.isRequired,
  statistics: PropTypes.object,
};

VideoInfo.defaultProps = {
  statistics: undefined
};