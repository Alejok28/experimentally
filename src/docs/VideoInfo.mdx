---
name: VideoInfo
route: /components/VideoInfo
menu: Components
---

import { Playground } from 'docz';
import VideoInfo from '../components/VideoInfo';
import { videos } from '../_test_/__fixtures__/videos';
import { statistics } from '../_test_/__fixtures__/statistics';

# VideoInfo
VideoInfo is a component to play a video using a url from YouTube. This component uses a react-player package 

## Properties

| Name          | Type     | Required     |Description                                |
|:--------------|:--------:|:------------:|:------------------------------------------|
| videoId       | String   | true         | Sting with the video Id to create the url of youtube |


## Basic usage

<Playground>
  <VideoInfo snippet={videos[0].snippet} statistics={statistics} />
</Playground>

