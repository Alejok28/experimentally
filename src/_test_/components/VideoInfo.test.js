import React from 'react';
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import VideoInfo from '../../components/VideoInfo'
import { videos } from '../__fixtures__/videos';
import { statistics } from '../__fixtures__/statistics';
import { escapeHtml } from '../../utils/functions';

afterEach(cleanup);

test('renders VideoInfo correctly', () => {
  const { container, getByText } = render(
    <VideoInfo snippet={videos[0].snippet} statistics={statistics} />
  );
  const title = getByText(escapeHtml(videos[0].snippet.title));
  const description = getByText(videos[0].snippet.description);
  const icons = container.querySelectorAll('svg');

  expect(icons).toHaveLength(3);
  expect(title).toBeInTheDocument();
  expect(description).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

test('renders VideoInfo without statistics', () => {
  const { container, getByText } = render(<VideoInfo snippet={videos[0].snippet} />);
  const title = getByText(escapeHtml(videos[0].snippet.title));
  const description = getByText(videos[0].snippet.description);

  expect(title).toBeInTheDocument();
  expect(description).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});