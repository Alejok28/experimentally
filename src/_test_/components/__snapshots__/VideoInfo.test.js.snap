// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VideoInfo correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Ryan Reynolds & Jake Gyllenhaal Answer the Web's Most Searched Questions | WIRED
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        'Life' stars Ryan Reynolds and Jake Gyllenhaal waste some time with WIRED and answer the Internet's most searched questions about themselves. Does Ryan ...
      </p>
    </div>
    <div
      class="MuiCardActions-root makeStyles-actions-3 MuiCardActions-spacing"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2z"
        />
      </svg>
      556518
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorError"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 3H6c-.83 0-1.54.5-1.84 1.22l-3.02 7.05c-.09.23-.14.47-.14.73v2c0 1.1.9 2 2 2h6.31l-.95 4.57-.03.32c0 .41.17.79.44 1.06L9.83 23l6.59-6.59c.36-.36.58-.86.58-1.41V5c0-1.1-.9-2-2-2zm4 0v12h4V3h-4z"
        />
      </svg>
      6249
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorDisabled"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
        />
      </svg>
      12663
    </div>
  </div>
</div>
`;

exports[`renders VideoInfo without statistics 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Ryan Reynolds & Jake Gyllenhaal Answer the Web's Most Searched Questions | WIRED
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        'Life' stars Ryan Reynolds and Jake Gyllenhaal waste some time with WIRED and answer the Internet's most searched questions about themselves. Does Ryan ...
      </p>
    </div>
  </div>
</div>
`;
